app/src/main/java/org/listenbrainz/android/data/repository/YimRepository.kt

```diff
@@ -5,4 +5,8 @@ import org.listenbrainz.android.util.Resource
 
 interface YimRepository {
     suspend fun getYimData(username: String): Resource<YimData>
+    
+    fun getUsername(): String?
```

Comment:
We shouldn't fetch the username again, but pass it as a parameter to functions from the UI.

Reply:
This is what I was talking about when i said move shared prefs to a separate repository. What I will do is make a complete new repositories which will contain similar prefs. This is done as to make testing flexible. Also, since shared prefs are a kind of "source of truth" or data. I feel it should go to repository. I'll be refactoring the whole code for this dw about that.

Reply:
I agree with the point that shared prefs can go into another repository. We should make a viewmodel for local storage and fetch the username and login status through it and feed it through the ui to the YIMViewModel. I am happy to wait for that inclusion in this PR.

Reply:
We don't need a view-model for this. Injecting a repository that has shared Preferences on any view-model (that needs it) would solve our issue. Since shared preferences have no requirement of a lifecycle, repository would do! 
Whatcha say?